addons/calendar/models/calendar_recurrence.py

```diff
@@ -129,38 +129,52 @@ class RecurrenceRule(models.Model):
          "The day must be between 1 and 31"),
     ]
 
+    def get_recurrence_name(self, recurrence, lang=None):
```

Comment:
you should remove `recurrence` and use self instead

also remove lang and just use self.env.lang in any case